### PR TITLE
fix(autoware_pid_longitudinal_controller): correct the PID controller

### PR DESCRIPTION
**Pull request by HansOersted**
## Description

The previous PID controller in Autoware is actually equivalent to a PD/PI controller.
This PR corrects the utilization of and upgrades to the PID controller.

**N.B. Without specific reasons, the parameters adopted in the previous PD/PI controller are adopted in [another PR](https://github.com/autowarefoundation/autoware_launch/pull/1083). And the parameters in that PR are equivalent to the PD/PI controller.**

**Before change:**
[Screencast from 07-19-2024 02_05_57 PM.webm](https://github.com/user-attachments/assets/2031b80b-a813-4b93-8398-77013562c9e7)

**After change:**
[Screencast from 07-19-2024 02_37_49 PM.webm](https://github.com/user-attachments/assets/3e0602fd-ce76-4abf-81a1-323c9931e1ea)

---

Note that one of the benefits of using implementing PID controllers is being robust to the external noise (e.g., driving on an ice road). Therefore, a noise was introduced artificially to test the PD/PI controller (before) and PID controller (after). 

**Noise definition**: a constant acceleration (+1.0 m/s2), that is `return feedback_acc + 1.0;`.
Also, note the PID output constraint has been relaxted to `max_out: 2.0`, `min_out: -2.0` in the analysis.

**Before change:**
[Screencast from 07-19-2024 02_15_43 PM.webm](https://github.com/user-attachments/assets/15116f58-1373-422a-a481-23a9b6ea6a69)

**After change (with new parameters below)**
```
kp: 1.5 # kp > 0 (stability proof) for third-order
ki: 0.8 # ki > 0 && Ki < kp * kd (stability proof) for third-order
kd: 2.5 # kd > 0 (stability proof) for third-order
```
[Screencast from 07-19-2024 04_26_14 PM.webm](https://github.com/user-attachments/assets/23b87cbd-15d2-4cad-a79d-3c5219481951)

**N.B. In the parameter PR, we picked the parameters equivalent to the PI/PD controller.**
It means that it will also show limit robustness to the external noise.

---

**Conclusions:**

1. The previous PI/PD controller shows the [logical error ](https://github.com/HansOersted/Autonomous_PID_controller_parameters), which is fixed in this PR.
2. The updated PID controller (this PR) works as expected in the normal case.
3. The previous PI/PD controller can fail to track the reference once the external disturbance in introduced.
4. The previous PI/PD controller successfully tracks the reference for some introduced external disturbances.

---

**Further steps:**

1. The user may change the coefficients of the PID controller based on the concentrations and missions.
2. The max P, I, and D constraints can be adjusted.
3. The coefficients of the PID controller for a robust tracking performances can be analyzed.

---

**Discussions:**

1. Honestly, erasing the integration when and only when the car is controlled manually is personally preferred. If latecomers have interests on this topic, do not hesitate to contact me.
2. If you are trying to implement Discussion 1, consider using `reset()`.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kyoichi-sugahara** [comment]
https://github.com/autowarefoundation/autoware.universe/blob/c0f07c2713c3bb53f915c86646293846263d6f1d/control/autoware_pid_longitudinal_controller/include/autoware/pid_longitudinal_controller/pid.hpp#L33
doxygen comment also should be changed

**Comment by HansOersted** [comment]
> https://github.com/autowarefoundation/autoware.universe/blob/c0f07c2713c3bb53f915c86646293846263d6f1d/control/autoware_pid_longitudinal_controller/include/autoware/pid_longitudinal_controller/pid.hpp#L33
> 
> 
> doxygen comment also should be changed

fixed in 55f0787dd9aa538f4f9b9a91591aa5d2f3ee48a6

File: control/autoware_pid_longitudinal_controller/src/pid_longitudinal_controller.cpp [hunk 1]
```diff
@@ -1107,11 +1107,13 @@ double PidLongitudinalController::applyVelocityFeedback(const ControlData & cont
     (vehicle_is_moving || (m_enable_integration_at_low_speed && vehicle_is_stuck)) &&
     is_under_control;
 
+  const bool erase_integral = !enable_integration;
```
Comment (takayuki5168) [inline control/autoware_pid_longitudinal_controller/src/pid_longitudinal_controller.cpp:1110]: Any reason to use the negative enable_integration here?
Using enable_integration consistently is easier to understand the code.

Comment (HansOersted) [reply control/autoware_pid_longitudinal_controller/src/pid_longitudinal_controller.cpp]: fixed in 0555ea54cec056d80840515a5a92ee328643a93b

**Comment by mitsudome-r** [comment]
@HansOersted @takayuki5168 Do you still need this modification?

**Comment by takayuki5168** [comment]
It will take time to confirm this PR including the real vehicle experiment and we don't have a plan for that, so let me close it for now.

**Pull request closed without merging**